documentation/smg/17_pixel_data.md

```diff
@@ -0,0 +1,427 @@
+for each of the nine fields of pixel data.
+There are also dependent properties for common groups of properties,
+e.g. `.coordinates` retrieves `u1`, `u2`, `u3` and `dE`.
+These properties can be used to retrieve and set data _in the current page_.
+
+```matlab
+>> size(pix.signal)
+    ans =
+      [1, pix.num_pixels]
+
+>> head_run_idx = pix.run_idx(1:10);
+>> size(head_run_idx)
+    ans =
+      [1, 10]
+
+>> tail_coordinates = pix.coordinates((end - 10):end);
+>> size(tail_coordinates)
+    ans =
+      [4, 10]
+```
+
+All these properties work by slicing into the cached data array at the correct
+column.
+Attempting to get/set pixels outside the range of the current page
+will throw an error.
+
+All properties use [_relative indexing_](#Absolute-and-Relative-Indexing).
+
+### Methods
+
+Getter and setter methods are also provided.
+
+```matlab
+>> coordinates = pix.get_data('coordinates');
+>> class(coordinates)
+    ans =
+      'double'
+
+>> pix.set_data({'signal', 'variance'}, zeros(2, pix.num_pixels));
+>> size(pix.get_data({'signal', 'variance'}, 1:100))
+    ans =
+      [2, 100]
+
+>> sub_pix = pix.get_pixels(3001:4000);
+>> class(sub_pix)
+    ans =
+      'PixelData'
+>> sub_pix.num_pixels
+    ans =
+      1000
+```
+
+These methods provide ways to retrieve data from the full pixel data array,
+whether that data is in memory or is in a file.
+The indexing argument attempts to mimic that of a normal Matlab array,
+i.e. logical indices are allowed and indices can be given out of order.
+However, it does not support indexing using `end`,
+you can use the `num_pixels` property to achieve the same effect.
+
+The getter methods have the advantage of being able to get/set multiple columns
+at once.
+This can be more efficient, as you're performing a single slice, instead of
+multiple.
+
+```matlab
+% The below examples are equivalent for a pixel data object with one page.
+% Using the get_data method is more efficient as it performs fewer slices
+
+% Performs one slice over the pixel array
+>> run_sig_var = pix.get_data({'run_idx', 'signal', 'variance'});
+
+% Performs one slice for each property access
+>> run_sig_var = zeros(3, pix.num_pixels);
+>> run_sig_var(1, :) = pix.run_idx;
+>> run_sig_var(2, :) = pix.signal;
+>> run_sig_var(3, :) = pix.variance;
+```
+
+All getter/setter methods use [_absolute indexing_](#Absolute-and-Relative-Indexing).
+
+#### Absolute and Relative Indexing
+
+Absolute indexing refers to indexing into the full pixel array.
+Relative indexing refers to indexing into the current page of pixels.
+
+So, suppose a `PixelData` object interfaces a file which contains 100 pixels,
+with a page size of 10 pixels.
+Then the following holds:
+
+```matlab
+>> pix.num_pixels
+     ans =
+       100
+>> pix.move_to_page(3);
+>> all(pix.signal(1:6) == pix.get_data('signal', 21:26))
+     ans =
+       true
+```
+
+There exists a private utility function that can perform the conversion between
```

Comment:
which function?

Reply:
I was reluctant to put the name of the function here, since it's a private method and the name could change. If you think it would be better to include the name, I'm happy to do so